src/monitor/process_monitor_main.cpp

```diff
@@ -53,6 +55,72 @@ namespace lo2s
 namespace monitor
 {
 
+static void drop_privileges()
+{
+    // get original uid & gid
+    gid_t original_uid;
+    uid_t original_gid;
+
+    if (config().user != "")
+    {
+        auto orig_user = getpwnam(config().user.c_str());
+        if (orig_user == nullptr)
+        {
+            Log::error() << "No user found with username '" + config().user + ".";
+            throw_errno();
+        }
+
+        original_uid = orig_user->pw_uid;
+        original_gid = orig_user->pw_gid;
+    }
+    else
+    {
+        // sudo is garanteed in this case, see check in config.hpp
+        try
+        {
+            original_uid = std::stoi(std::getenv("SUDO_UID"));
+            original_gid = std::stoi(std::getenv("SUDO_GID"));
+        }
+        catch (std::invalid_argument const& e)
+        {
+            Log::error() << "Cannot parse SUDO_UID/SUDO_GID into int.";
```

Comment:
This will be a user-facing message. Users don't know what an `int` is or that `SUDO_UID/SUDO_GID` are environment variables.

`Cannot parse the environment variables SUDO_UID and/or SUDO_GID.`